spec/lang/step/operators.md

```diff
@@ -234,7 +234,7 @@ impl<M: Memory> Machine<M> {
         match val {
             Value::Int(i) => i,
             Value::Bool(b) => if b { Int::from(1) } else { Int::from(0) },
-            Value::Ptr(p) => p.addr,
+            Value::Ptr(p) => p.thin_pointer.addr,
```

Comment:
This just ignores the metadata entirely for comparisons... whereas in rustc comparison behaves as-if this was a tuple. So we'll have to do something with the metadata. For now we can at least assert that it is None to ensure we notice if this ever goes wrong.

Reply:
Yes this is quite poorly documented. And indeed, it is lexicographically sorted.

For now, I think you should make well-formed reject and wide pointer passed to this operand, and then add a panic if `p` has metadata.